sw/device/tests/aes_force_prng_reseed_test.c

```diff
@@ -0,0 +1,264 @@
+  CHECK_DIF_OK(dif_edn_get_status(edn1, kDifEdnStatusReady, &edn_status1));
+  LOG_INFO("EDN1 status (Ready): 0x%x", edn_status1);
+}
+
+// AES key used for the test
+static const unsigned char kAesModesKey128[16] = {
+    0x2b, 0x7e, 0x15, 0x16, 0x28, 0xae, 0xd2, 0xa6,
+    0xab, 0xf7, 0x15, 0x88, 0x09, 0xcf, 0x4f, 0x3c};
+
+static const unsigned char kAesModesPlainText[16] = {
+    0x6b, 0xc1, 0xbe, 0xe2, 0x2e, 0x40, 0x9f, 0x96,
+    0xe9, 0x3d, 0x7e, 0x11, 0x73, 0x93, 0x17, 0x2a,
+};
+
+static const unsigned char kAesModesCipherTextEcb128[16] = {
+    0x3a, 0xd7, 0x7b, 0xb4, 0x0d, 0x7a, 0x36, 0x60,
+    0xa8, 0x9e, 0xca, 0xf3, 0x24, 0x66, 0xef, 0x97};
+
+/*
+The mask share, used to mask kKey. Note that the masking should not be done
+manually. Software is expected to get the key in two shares right from the
+beginning.
+*/
+static const uint8_t kKeyShare1[] = {
+    0x0f, 0x1f, 0x2f, 0x3F, 0x4f, 0x5f, 0x6f, 0x7f, 0x8f, 0x9f, 0xaf,
+    0xbf, 0xcf, 0xdf, 0xef, 0xff, 0x0a, 0x1a, 0x2a, 0x3a, 0x4a, 0x5a,
+    0x6a, 0x7a, 0x8a, 0x9a, 0xaa, 0xba, 0xca, 0xda, 0xea, 0xfa,
+};
+
+// Function to print AES blocks key and data
+static void aes_print_block(const unsigned char *data, const int num_bytes) {
+  for (int i = 0; i < num_bytes; i++) {
+    LOG_INFO("%02x ", data[i]);
+  }
+  LOG_INFO("");
+}
+
+status_t execute_test(void) {
+  // Initialize AES, EDN, CSRNG, and Entropy Source
+
+  // Initialize all modules
+  LOG_INFO("Initializing AES unit.");
+  CHECK_DIF_OK(
+      dif_aes_init(mmio_region_from_addr(TOP_EARLGREY_AES_BASE_ADDR), &aes));
+  CHECK_DIF_OK(dif_aes_reset(&aes));
+  LOG_INFO("Module initialized: AES");
+  LOG_INFO("Starting AES PRNG reseed stall test with extra debug...");
+  CHECK_DIF_OK(
+      dif_edn_init(mmio_region_from_addr(TOP_EARLGREY_EDN0_BASE_ADDR), &edn0));
+  CHECK_DIF_OK(
+      dif_edn_init(mmio_region_from_addr(TOP_EARLGREY_EDN1_BASE_ADDR), &edn1));
+  CHECK_DIF_OK(dif_csrng_init(
+      mmio_region_from_addr(TOP_EARLGREY_CSRNG_BASE_ADDR), &csrng));
+  CHECK_DIF_OK(dif_entropy_src_init(
+      mmio_region_from_addr(TOP_EARLGREY_ENTROPY_SRC_BASE_ADDR), &entropy_src));
+  LOG_INFO("Modules initialized: EDN0, EDN1, CSRNG, Entropy Source");
+
+  // Configure AES for ECB mode, 128-bit key, and manual operation
+  dif_aes_transaction_t transaction = {
+      .operation = kDifAesOperationEncrypt,
+      .mode = kDifAesModeEcb,
+      .key_len = kDifAesKey128,
+      .key_provider = kDifAesKeySoftwareProvided,
+      .mask_reseeding = kDifAesReseedPerBlock,
+      .manual_operation = kDifAesManualOperationManual,
+      .reseed_on_key_change = true,
+      .force_masks = false,
+      .ctrl_aux_lock = false,
+  };
+
+  LOG_INFO(
+      "AES transaction configured: ECB mode, 128-bit key, manual operation.");
+
+  // Disable EDN and CSRNG to simulate stalling
+  LOG_INFO("Disabling EDN0, EDN1...");
+  CHECK_DIF_OK(dif_edn_stop(&edn0));
+  CHECK_DIF_OK(dif_edn_stop(&edn1));
+  busy_spin_micros(500);
```

Comment:
Use the following function from `entropy_testutils`:

```c
TRY(entropy_testutils_stop_all());
```

Reply:
Noted